src/menus.ts

```diff
@@ -103,25 +103,25 @@ export class PopupMenu extends Menu {
         return false;   // block all keys other than ours
     }
 
-    searchFor(match: string) {
-        const parts = match.split("").map(escapeRegex);
-        return (
-            this.find(new RegExp("^"+ parts.join(""), "ui")) ||
-            this.find(new RegExp("^"+ parts.join(".*"), "ui")) ||
-            this.find(new RegExp(parts.join(".*"), "ui"))
-        );
-    }
-
-    find(pattern: RegExp) {
-        let pos = Math.min(0, this.selected);
+    searchFor(match: string): boolean {
+        const query = prepareQuery(match);
+        let pos = Math.min(0, this.selected), 
+            results = [] as (SearchResultContainer & { pos: number })[];
         for (let i=this.items.length; i; ++pos, i--) {
             if (this.items[pos]?.disabled) continue;
-            if (this.items[pos]?.dom.textContent.match(pattern)) {
```

Comment:
Is this intentional to get all texts, including extensions located in `.nav-file-tag`?